mw-chat-bff/internal/controllers/rooms.controller.go

```diff
@@ -65,9 +94,36 @@ func (cc *RoomsController) GetRooms(ctx *gin.Context) {
 func (cc *RoomsController) GetRoomById(ctx *gin.Context) {
 	roomId := ctx.Param("roomId")
 
-	room, err := cc.RoomsService.GetRoomById(ctx, roomId)
+	room, err := cc.roomsService.GetRoomById(ctx, roomId)
 	util.HandleErrorGin(ctx, err)
 
+	userIDs := lo.Map(room.Users, func(user schemas.UserResponse, _ int) string {
+		return user.UserID
+	})
+
+	populatedUserMap, err := cc.usersService.GetChatUsers(ctx, userIDs)
+	util.HandleErrorGin(ctx, err)
+
+	room.Users = lo.Map(room.Users, func(user schemas.UserResponse, _ int) schemas.UserResponse {
+		if populatedUser, ok := populatedUserMap[user.UserID]; ok {
+			user.Name = populatedUser.Name
+			user.ImageURL = populatedUser.ImageURL
+		}
+		return user
+	})
+
+	room.Messages = lo.Map(room.Messages, func(message schemas.MessageResponse, _ int) schemas.MessageResponse {
+		if populatedUser, ok := populatedUserMap[message.OwnerID]; ok {
+			message.OwnerName = populatedUser.Name
+			message.OwnerImageURL = populatedUser.ImageURL
+		}
+		return message
+	})
+
+	if room.RoomType == "private" {
```

Comment:
I added an enum to the Swagger description for the GetRooms function in rooms.controller.go, regenerated Swagger and OpenAPI, but no values appeared that I could use. Unfortunately, I couldn’t find a ready-made solution online. I decided it would be better to get rid of the strings, and added two constants for private and group.